app/code/core/Mage/Newsletter/controllers/SubscriberController.php

```diff
@@ -98,8 +98,11 @@ public function confirmAction()
             $subscriber = Mage::getModel('newsletter/subscriber')->load($id);
             $session = Mage::getSingleton('core/session');
 
-            if ($subscriber->getId() && $subscriber->getCode()) {
-                if ($subscriber->confirm($code)) {
+            if ($subscriber->getStatus() == $subscriber::STATUS_SUBSCRIBED){
+                $session->addNotice($this->__('This email address is already confirmed.'));
+            } elseif($subscriber->getId() && $subscriber->getCode()) {
+                if($subscriber->confirm($code)) {
+                    $status = $subscriber->sendConfirmationSuccessEmail();
```

Comment:
I'm gonna beat @luigifab to mention the space missing after `if` and `elseif`, and before the curly brace `{`. `$status` variable is also unused so it can be removed :)

Reply:
done! ;-)

Reply:
done? there are 2x `if(`

Reply:
😰 done now